tests/aggregation/tests.py

```diff
@@ -2359,3 +2390,256 @@ def test_aggregate_combined_queries(self):
             )
         )
         self.assertEqual(qs.count(), 3)
+
+
+class StringAggTestModel(Model):
+    char_field = CharField(max_length=30, blank=True)
+    text_field = TextField(blank=True)
+    json_field = JSONField(null=True)
+
+
+class StatTestModel(Model):
+    int1 = IntegerField()
+    int2 = IntegerField()
+    related_field = ForeignKey(StringAggTestModel, SET_NULL, null=True)
+
+
+class StringAggTestCase(TestCase):
+    @classmethod
+    def setUpTestData(cls):
+        cls.aggs = StringAggTestModel.objects.bulk_create(
+            [
+                StringAggTestModel(
+                    char_field="Foo1",
+                    text_field="Text1",
+                ),
+                StringAggTestModel(
+                    char_field="Foo2",
+                    text_field="Text2",
+                    json_field={"lang": "pl"},
+                ),
+                StringAggTestModel(
+                    char_field="Foo4",
+                    text_field="Text4",
+                    json_field={"lang": "en"},
+                ),
+                StringAggTestModel(
+                    char_field="Foo3",
+                    text_field="Text3",
+                    json_field={"breed": "collie"},
+                ),
+            ]
+        )
+
+    def test_empty_result_set(self):
+        StringAggTestModel.objects.all().delete()
+        aggregation = StringAgg("char_field", delimiter=";")
+
+        # Empty result with non-execution optimization.
+        with self.assertNumQueries(0):
+            values = StringAggTestModel.objects.none().aggregate(
+                aggregation=aggregation,
+            )
+            self.assertEqual(values, {"aggregation": None})
+        # Empty result when query must be executed.
+        with self.assertNumQueries(1):
+            values = StringAggTestModel.objects.aggregate(
+                aggregation=aggregation,
+            )
+            self.assertEqual(values, {"aggregation": None})
+
+    def test_default_argument(self):
+        StringAggTestModel.objects.all().delete()
+        tests = [
+            (StringAgg("char_field", delimiter=";", default="<empty>"), "<empty>"),
```

Comment:
I would expect this to crash as `default` must be passed as `Value`? Passing `str` to `Func` should normally attempt resolve to field references (e.g. `"<empty>" -> F("<empty>")") and crash if missing.

Reply:
> delimiter should not be wrapped in a Value as that prevent references to columns and normally string arguments are considered to be `F`'ield references.

The `Value` wrapper on a string comes right from the old implementation in the Postgres-specific version. Would it make more sense to allow for the string to come through and wrap it as a value for 5.2 with a deprecation warning in 6? That does break the ability to use a string as an alias for `F` for 5.2, though.

Reply:
I think I have something together that wraps the `__init__` from the Postgres version of the `StringAgg` class and does the `Value(str(delimiter))` and logs a deprecation warning. The main `StringAgg` can then expect an `Expression` of some sort.

Reply:
> I think I have something together that wraps the __init__ from the Postgres version of the StringAgg class and does the Value(str(delimiter)) and logs a deprecation warning. The main StringAgg can then expect an Expression of some sort.

That's perfect, using the deprecation period for `postgres.StringAgg` to `models.StringAgg` seems like a great way to normalize this behavior 🎉